src/cpp/ext/otel/otel_plugin.h

```diff
@@ -189,10 +189,30 @@ class OpenTelemetryPluginBuilderImpl {
   std::shared_ptr<std::set<absl::string_view>> optional_label_keys_;
 };
 
+class ActivePluginOptionsView;
 class OpenTelemetryCallTracer;
 
 class OpenTelemetryPlugin : public grpc_core::StatsPlugin {
  public:
+  class ScopeConfig : public grpc_core::StatsPlugin::ScopeConfig {
+   public:
+    ScopeConfig(
+        std::unique_ptr<ActivePluginOptionsView> active_plugin_options_view,
+        absl::string_view filtered_target)
+        : active_plugin_options_view_(std::move(active_plugin_options_view)),
+          filtered_target_(filtered_target) {}
+
+    ActivePluginOptionsView* active_plugin_options_view() const {
+      return active_plugin_options_view_.get();
+    }
+
+    absl::string_view filtered_target() const { return filtered_target_; }
+
+   private:
+    std::unique_ptr<ActivePluginOptionsView> active_plugin_options_view_;
```

Comment:
Why does this need to be dynamically allocated?  Why not simply inline it?

If the problem here is the declaration order, it seems like we could move the declaration of `ActivePluginOptionsView` up above this, and move the body of its methods into the .cc file.

Reply:
Done.